src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+          </ModalContext.Provider>
+        );
+      }).not.toThrow();
+    });
+
+    it("should not update focusable elements if wrapper ref isn't found", () => {
+      const wrapperRef = { current: null };
+      expect(() => {
+        render(
+          <ModalContext.Provider value={{ isAnimationComplete: true }}>
+            <FocusTrap wrapperRef={wrapperRef}>
+              <div id="myComponent">Content</div>
+            </FocusTrap>
+          </ModalContext.Provider>
+        );
+      }).not.toThrow();
+    });
+  });
+
+  describe("additionalWrapperRefs", () => {
+    const BUTTON_IN_WRAPPER = "Button in wrapper";
+    const BUTTON_IN_ADDITIONAL_WRAPPER_ONE = "Button in additional wrapper one";
+    const BUTTON_IN_ADDITIONAL_WRAPPER_TWO = "Button in additional wrapper two";
+
+    beforeEach(() => {
+      const wrapperRef = { current: null };
+      const otherContentRef1 = { current: null };
+      const otherContentRef2 = { current: null };
+
+      render(
+        <ModalContext.Provider value={{ isAnimationComplete: true }}>
+          <FocusTrap
+            wrapperRef={wrapperRef}
+            additionalWrapperRefs={[otherContentRef1, otherContentRef2]}
+          >
+            <div data-testid={WRAPPER_ID}>
+              <button type="button" id="outside1">
+                outside focus trap
+              </button>
+              <div ref={wrapperRef}>
+                <button type="button" id="insidewrapper">
+                  {BUTTON_IN_WRAPPER}
+                </button>
+              </div>
+              <button type="button" id="outside2">
+                outside focus trap
+              </button>
+              <div ref={otherContentRef1}>
+                <button type="button" id="insideother1">
+                  {BUTTON_IN_ADDITIONAL_WRAPPER_ONE}
+                </button>
+              </div>
+              <button type="button" id="outside3">
+                outside focus trap
+              </button>
+              <div ref={otherContentRef2}>
+                <button type="button" id="insideother2">
+                  {BUTTON_IN_ADDITIONAL_WRAPPER_TWO}
+                </button>
+              </div>
+              <button type="button" id="outside4">
+                outside focus trap
+              </button>
+            </div>
+          </FocusTrap>
+        </ModalContext.Provider>
+      );
+    });
+
+    it("tab should cycle through focusable elements inside the provided container refs and ignore all others", () => {
+      focusElement(screen.getByText(BUTTON_IN_WRAPPER));
+      tabPress();
+      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_ONE)).toBeFocused();
+      tabPress();
+      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_TWO)).toBeFocused();
+      tabPress();
+      expect(screen.getByText(BUTTON_IN_WRAPPER)).toBeFocused();
+    });
+  });
+
+  describe("when content in the children tree changes", () => {
+    it("should trigger the MutationObserver", () => {
```

Comment:
We might be able to workaround testing the implementation details of `MutationObserver` here by using some of the [async wait utilities part of RTL](https://testing-library.com/docs/guide-disappearance#waiting-for-appearance) to wait until `children` changes and verify if the button is then focused. That way we could mimic what the user would see. Though we may still need to write a manual jest mock for `MutationObserver` like you've done here, since I can't seem to find any information that `js-dom` simulates it for us. What do you think?

If this isn't feasible, perhaps it would be worth us in a separate ticket creating a manual jest mock for `MutationObserver` inside `src/__spec_helper__` anyway, like we have done for `matchMedia` and `ResizeObserver`, to help keep things consistent?

Reply:
I don't think there's a need to write a mock for the MutationObserver as this is the only file that uses it (if that changes it will be worthwhile) and I think we should really be testing this behaviour in a browser via cypress but we have coverage constraints. The coverage is actually met elsewhere for this but I decided to keep this test as it will let anyone working on it know that it has failed straightaway rather than catching it via missing coverage etc. I do agree your suggestion is a bit tidier than what we originally had so I'll update it a remove the mocking of the observer.